system/datatypes/mailinglist.php

```diff
@@ -781,52 +817,111 @@ private function testSendMailing($mailingId, $email) {
 	 *  implement if option (2) message with 'do not close client'
 	 *  store date with locale
 	 *
-	 * @param string $mailingId
+	 * @param HyphaRequest $request
 	 *
 	 * @throws Exception
+	 * @return null
 	 */
-	private function sendMailing($mailingId) {
+	private function mailingSendAction(HyphaRequest $request) {
 		if (!$this->hasSender()) {
-			throw new \Exception(__('ml-no-sender'));
+			notify('error', __('login-to-edit'));
```

Comment:
This mixes up two checks: `!hasSender()` now shows `login-to-edit` (which is wrong), and there is no actual `!isUser()` check (seems it was not there before either).

Reply:
I blame my laziness.

Reply:
fixed it

Reply:
thank you for catching this. :-)